src/components/Badge/Badge.test.tsx

```diff
@@ -0,0 +1,20 @@
+import { render, screen } from '@testing-library/react';
+import '@testing-library/jest-dom';
+//import renderer from 'react-test-renderer';
+import { Badge } from './index';
+
+describe('Badge', () => {
+  it('render Badge with default text content', () => {
+    const text = 'Welcome to Badge Component';
+    render(<Badge size={24} name="star" color="red" />);
+    expect(screen.getByText(text)).toBeInTheDocument();
```

Comment:
use tailwind custom classes sm, md ,..etc and test for them in the document, 
e.g size = "sm"  => test case that looks for sm in the document, you don't need to cover all variations, 2 is enough